src/FieldWarning/Assets/Ingame/UI/InputManager.cs

```diff
@@ -30,388 +32,327 @@ namespace PFW.Ingame.UI
-
-            foreach (var s in pointList) {
-                if (Vector3.Distance(p, s.transform.position) < distance) {
-                    distance = Vector3.Distance(p, s.transform.position);
-                    go = s;
-                }
-            }
-            return go;
+            return this.IsMouse(events, button);
         }
 
-        public void AddSpawnPoint(SpawnPointBehaviour s)
+        private void MaybeExitPurchasingMode()
         {
-            if (!_spawnPointList.Contains(s))
-                _spawnPointList.Add(s);
+            if (!this.IsMouse(Events.Clicked))
+                return;
+            this._currentBuyTransaction.GhostPlatoons.ForEach(
+                ghost => ghost.GetComponent<GhostPlatoonBehaviour>().Destroy());
+            this.ExitPurchasingMode();
         }
 
-        public void ApplyHotkeys()
+        private void MaybePurchaseGhostUnits(SpawnPointBehaviour closestSpawn)
         {
-            if (Commands.Unload) {
-                _selectionManager.DispatchUnloadCommand();
-
-            } else if (Commands.Load) {
-                _selectionManager.DispatchLoadCommand();
+            if (closestSpawn is null || this.IsMouse(Events.Released)
+                                     || EventSystem.current.currentSelectedGameObject is null
+                                     || this._currentBuyTransaction is null)
+                return;
 
-            } else if (Commands.FirePos && !_selectionManager.Empty) {
-                EnterFirePosMode();
+            closestSpawn.BuyPlatoons(this._currentBuyTransaction.GhostPlatoons);
 
-            } else if (Commands.ReverseMove && !_selectionManager.Empty) {
-                EnterReverseMoveMode();
-
-            } else if (Commands.FastMove && !_selectionManager.Empty) {
-                EnterFastMoveMode();
-            }
+            if (KeyDown(KeyCode.LeftShift))
+                this._currentBuyTransaction = this._currentBuyTransaction.Clone();
+            else
+                this.ExitPurchasingMode();
         }
 
-        private void EnterFirePosMode()
+        private void MoveGhostsToMouse()
         {
-            CurMouseMode = MouseMode.firePos;
-            Vector2 hotspot = new Vector2(_firePosReticle.width / 2, _firePosReticle.height / 2);
-            Cursor.SetCursor(_firePosReticle, hotspot, CursorMode.Auto);
+            if (Util.GetTerrainClickLocation(out RaycastHit hit))
+                this._selectionManager.PrepareMoveOrderPreview(hit.point);
         }
 
-        private void EnterNormalMode()
+        private void OnOrderHold()
         {
-            CurMouseMode = MouseMode.normal;
-            Cursor.SetCursor(null, Vector2.zero, CursorMode.Auto);
+            if (Util.GetTerrainClickLocation(out RaycastHit hit))
+                this._selectionManager.RotateMoveOrderPreview(hit.point);
         }
 
-        private void EnterFastMoveMode()
+        private void SetCursor(Modes modes)
```

Comment:
This method does not just set the cursor. The old name of "Enter___Mode" reflected that. Also your enum naming confused you here, the `modes` variable actually only holds one mode at a time :D